DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -239,8 +249,29 @@ DlsSubApplication dlsSubApplication
                 delegateAccount?.Id
             );
 
+            var modifiedEmails = new List<(string, int?)>();
+            var shouldLogIntoCentrelessAccount = false;

```

Comment:
This is really `shouldRedirectToVerifyEmail && centreId != null`. It would be safer not to track it separately.